backend/routes/flightStatus.js

```diff
@@ -0,0 +1,46 @@
+import { API_KEY, API_SECRET } from "../config.js";
+import Amadeus from "amadeus";
+import express from "express";
+import { PrismaClient } from '@prisma/client'; 
+import { authenticateUser } from '../auth.js';
+
+const prisma = new PrismaClient();
+const router = express.Router();
+const amadeus = new Amadeus({
+    clientId: API_KEY,
+    clientSecret: API_SECRET,
+});
+
+//flight search for adding trip - needs authentication
+router.get('/flight-status', authenticateUser, async (req, res) => {
+    try {
+        const { carrierCode, flightNumber, scheduledDepartureDate } = req.query;
+        if (!carrierCode || !flightNumber || !scheduledDepartureDate) {
+            return res.status(400).json({ error: 'Missing required parameters' });
+        }
+        const response = await amadeus.schedule.flights.get({
+            carrierCode, flightNumber, scheduledDepartureDate
+        });
+        res.json(JSON.parse(response.body));
+    } catch (err) {
+        res.status(500).json({ error: 'An error occurred while fetching flight status', details: err.message });
+    }
+});
+
+
+// saving static trip details on adding trip - needs authentication
+router.post('/add-trip', authenticateUser, async (req, res) => { 
+    try { 
+        const { carrierCode, flightNumber, scheduledDepartureDate } = req.body; 
+        const userId = req.session.user.userId;
+        const newTrip = await prisma.trip.create({ 
+        data: { userId, carrierCode, flightNumber, scheduledDepartureDate } 
```

Comment:
So you won't store any data from the external API locally?

Reply:
Right now, I'm only saving the essentials like carrierCode, flightNumber, scheduledDeparture, and userID. This helps us quickly fetch flight statuses for the upcoming trips page without storing too much data upfront. Later I'll use it to fetch realtime data for upcoming pages.
Once the flight lands, we'll save the full details in a new table, maybe called PastTrips. This way, we can easily show all the past flights on the Past Trip page.
Let me know if you think we should tweak this approach or if there's anything else to consider!